### ISSUE TITLE
Compartment censors unsafe globals: these should effect ReferenceError on access

### ISSUE THREAD

**Issue by kriskowal** [endojs/endo#309]
From `ses/test/typeof.test.js`:

```js
  // TODO: the Compartment currently censors objects from the unsafe global, but
  // they appear defined as 'undefined' and don't throw a ReferenceError.
  t.doesNotThrow(() => c.evaluate('globalThis'), ReferenceError);
  t.equal(c.evaluate('globalThis'), undefined);
  t.equal(c.evaluate('typeof globalThis'), 'undefined');
```

**Comment by erights** [comment]
On the 8 magic lines of code, it is impossible for us to both throw a ReferenceError when we should for an unknown variable `foo` and not throw a ReferenceError for `typeof foo`. This would be fixed by #730

**Comment by kriskowal** [comment]
This cannot be solved by the shim. Closing and tracking what we can fix at #730.
